src/constraints/singlevariable.jl

```diff
@@ -40,11 +40,30 @@ function setvariablebound!(m::LinQuadOptimizer, v::SinVar, set::IV)
     setvariablebound!(m, getcol(m, v), set.lower, backend_type(m, Val{:Lowerbound}()))
 end
 
+SVCI(v::SinVar, ::S) where S = SVCI{S}(v.variable.value)
+
+function checkexisting(m::LinQuadOptimizer, v::SinVar, set::S) where S
+    ref = SVCI(v, set)
+    if haskey(constrdict(m, ref), ref)
+        error("Adding the same constraint type: $(S) is not allowed for SingleVariable function")
+    end
+end
+
+function checkconflicting(m::LinQuadOptimizer, v::SinVar, set_to_add::S0, set_to_test::S) where S where S0
+    ref = SVCI(v, set_to_test)
+    if haskey(constrdict(m, ref), ref)
+        error("Adding the same constraint type: $(S0) is not allowed for SingleVariable function because there is constraint of type $(S) tied to the respective variable")
+    end
+end
+
 # add constraint
 function MOI.addconstraint!(m::LinQuadOptimizer, v::SinVar, set::S) where S <: LinSets
+    checkexisting(m, v, set)
+    checkconflicting(m, v, set, MOI.Semicontinuous(0.0, 0.0))
+    checkconflicting(m, v, set, MOI.Semiinteger(0.0, 0.0))
+    checkconflicting(m, v, set, MOI.ZeroOne())
     setvariablebound!(m, v, set)
-    m.last_constraint_reference += 1
-    ref = SVCI{S}(m.last_constraint_reference)
+    ref = SVCI(v, set)
```

Comment:
Does this impact the requirement to return constraints in the order that they were created?
https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/11fb079aa1a3b63a21f49141671ee1b5aa0392e8/src/constraints.jl#L85

Reply:
Yes it does.
I am proposing that this rule is not applied to Single variable constraints since we only have one of each - since they are tied to the variable reference they will be unique...

Reply:
the well ordering on variables will keep these in some fixed order

Reply:
I see the issue with http://www.juliaopt.org/MathOptInterface.jl/latest/apireference.html#MathOptInterface.ListOfConstraintIndices. We definitely want the constraint indices to be tied to the variable indices for bound-type constraints. On the other hand, we should be consistent with how constraints work across all classes of constraints. What about using an `OrderedSet` to maintain the list of constraints in the order they're added?